### ISSUE TITLE
INAV FW 1.8.1 GPS no Lock

### ISSUE THREAD

**Issue by BadgerG** [iNavFlight/inav#2871]
I have an SP Racing F3, most likely a clone. I have been fighting a GPS issue for a couple of weeks. I have tried 3 different SP Racing F3 boards and 3 different GPS units with similar results. Here is my  setup:

SP Racing F3 FC
INAV 1.8.1 
FRSky  X8R receiver (SBUS connection, UART 3)
HolyBro M8N GPS module with compass (UART0)
3DR Telemetry radio (915 MHz, UART2)
FRSky X9D Transmitter
King Kong 260 Quad frame

I can plug the battery in and it takes 3 to 4 minutes with the quad by a window and the GPS modules blue lock light starts flashing. The INAV configurator (1.9) shows no indication of GPS lock (GPS icon is red). Then the quad will sound a lock tone and it will have GPS with anywhere from 7 to 12 sats.

Then a few seconds later it loses lock and the icon turns red again. The GPS still show a blinking blue light. Then every 1 or 2 minutes it will get lock for a few more second 10-15 seconds at the most. Cycling power on the quad fixes nothing. I can at that point take it outside and I get the exact same result accept it will show more satellites are locked. At times I can see it going from lock to no lock and back repeatedly even though the GPS module show a blue locked light.

I can take the same GPS unit and connect it to my Tarot 680 Pro with a Pixhawk and within 15 seconds it shows lock with 11+ satellites and never loses lock. It normally shows 13 to 18 satellites locked on the original GPS unit and this was able to lock on to 16 in the house. The other GPS modules act the same way on the SPR F3 and get significantly fast lock and sat number on the Pixhawk. So in this case this is not a problem with the GPS modules or the SPR F3.

I can arm the quad and fly it after turning off the extra arming checks, and it flies pretty well. Alt hold works but need tuning.

So to me in mho, is that the GPS code is not getting or using the information the GPS unit is sending or the configuration that is sent to the GPS is not the same as what the Pixhawk sends to the GPS module. So far I have never been able to get any GPS units to work anywhere near as good under INAV and APM.

Because of the way my ports are setup I have to disable the GPS and Mag in order to connect to the USB to get blackbox logs or settings dump, so I don’t have them. 

One other thing is that I am using and external Mag on the GPS module and the mag appears to work just fine, with no I2C errors. I cut the trace to the internal mag. I did a compass calibration and it was successful. The Quad show good health except for the GPS not being locked. If this was just one FC board and 1 GPS unit I would say faulty hardware, but these are brand new FC and GPS modules.

At first I thought maybe the UART port had issues, but it works just fine for installing firmware to the boards at 115K baud so it is working fine. I really don’t want to point the finger at the firmware, but I don’t know what else could be causing perfectly good hardware not to work that works fine using different firmware on a different board.

This happens in INAV 1.9 and Mission planner for INAV.

**Comment by shellixyz** [comment]
Please post a dump of your settings

**Comment by BadgerG** [comment]
Okay I pull the quad apart so I can connect to the USB port to down load the dump.

# version
# INAV/SPRACINGF3 1.7.0 May  3 2017 / 11:12:57 (104b66c)

# resources

# mixer
mixer QUADX

mmix reset


# servo
servo 0 1000 2000 1500 100 -1 
servo 1 1000 2000 1500 100 -1 
servo 2 1000 2000 1500 100 -1 
servo 3 1000 2000 1500 100 -1 
servo 4 1000 2000 1500 100 -1 
servo 5 1000 2000 1500 100 -1 
servo 6 1000 2000 1500 100 -1 
servo 7 1000 2000 1500 100 -1 

# servo mix
smix reset


# feature
feature -RX_PPM
feature -VBAT
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -TELEMETRY
feature -CURRENT_METER
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DASHBOARD
feature -BLACKBOX
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -SUPEREXPO
feature -VTX
feature -RX_SPI
feature -SOFTSPI
feature -PWM_SERVO_DRIVER
feature -PWM_OUTPUT_ENABLE
feature -OSD
feature VBAT
feature RX_SERIAL
feature GPS
feature TELEMETRY
feature BLACKBOX
feature PWM_OUTPUT_ENABLE

# beeper
beeper GYRO_CALIBRATED
beeper -HW_FAILURE
beeper RX_LOST
beeper RX_LOST_LANDING
beeper DISARMING
beeper ARMING
beeper ARMING_GPS_FIX
beeper BAT_CRIT_LOW
beeper BAT_LOW
beeper GPS_STATUS
beeper RX_SET
beeper ACTION_SUCCESS
beeper ACTION_FAIL
beeper READY_BEEP
beeper MULTI_BEEPS
beeper DISARM_REPEAT
beeper ARMED
beeper SYSTEM_INIT
beeper ON_USB
beeper LAUNCH_MODE

# map
map AETR1234

# serial
serial 0 1 115200 38400 0 115200
serial 1 1 57600 38400 0 115200
serial 2 64 115200 38400 0 115200

# led
led 0 0,0::C:0
led 1 0,0::C:0
led 2 0,0::C:0
led 3 0,0::C:0
led 4 0,0::C:0
led 5 0,0::C:0
led 6 0,0::C:0
led 7 0,0::C:0
led 8 0,0::C:0
led 9 0,0::C:0
led 10 0,0::C:0
led 11 0,0::C:0
led 12 0,0::C:0
led 13 0,0::C:0
led 14 0,0::C:0
led 15 0,0::C:0
led 16 0,0::C:0
led 17 0,0::C:0
led 18 0,0::C:0
led 19 0,0::C:0
led 20 0,0::C:0
led 21 0,0::C:0
led 22 0,0::C:0
led 23 0,0::C:0
led 24 0,0::C:0
led 25 0,0::C:0
led 26 0,0::C:0
led 27 0,0::C:0
led 28 0,0::C:0
led 29 0,0::C:0
led 30 0,0::C:0
led 31 0,0::C:0

# color
color 0 0,0,0
color 1 0,255,255
color 2 0,0,255
color 3 30,0,255
color 4 60,0,255
color 5 90,0,255
color 6 120,0,255
color 7 150,0,255
color 8 180,0,255
color 9 210,0,255
color 10 240,0,255
color 11 270,0,255
color 12 300,0,255
color 13 330,0,255
color 14 0,0,0
color 15 0,0,0

# mode_color
mode_color 0 0 1
mode_color 0 1 11
mode_color 0 2 2
mode_color 0 3 13
mode_color 0 4 10
mode_color 0 5 3
mode_color 1 0 5
mode_color 1 1 11
mode_color 1 2 3
mode_color 1 3 13
mode_color 1 4 10
mode_color 1 5 3
mode_color 2 0 10
mode_color 2 1 11
mode_color 2 2 4
mode_color 2 3 13
mode_color 2 4 10
mode_color 2 5 3
mode_color 3 0 8
mode_color 3 1 11
mode_color 3 2 4
mode_color 3 3 13
mode_color 3 4 10
mode_color 3 5 3
mode_color 4 0 7
mode_color 4 1 11
mode_color 4 2 3
mode_color 4 3 13
mode_color 4 4 10
mode_color 4 5 3
mode_color 5 0 9
mode_color 5 1 11
mode_color 5 2 2
mode_color 5 3 13
mode_color 5 4 10
mode_color 5 5 3
mode_color 6 0 6
mode_color 6 1 10
mode_color 6 2 1
mode_color 6 3 0
mode_color 6 4 0
mode_color 6 5 2
mode_color 6 6 3
mode_color 6 7 6
mode_color 6 8 0
mode_color 6 9 0
mode_color 6 10 0

# aux
aux 0 0 1 1300 2100
aux 1 1 0 1325 1600
aux 2 2 0 1850 2100
aux 3 20 1 1800 2100
aux 4 5 3 1300 1700
aux 5 0 0 900 900
aux 6 0 0 900 900
aux 7 0 0 900 900
aux 8 0 0 900 900
aux 9 0 0 900 900
aux 10 0 0 900 900
aux 11 0 0 900 900
aux 12 0 0 900 900
aux 13 0 0 900 900
aux 14 0 0 900 900
aux 15 0 0 900 900
aux 16 0 0 900 900
aux 17 0 0 900 900
aux 18 0 0 900 900
aux 19 0 0 900 900

# adjrange
adjrange 0 0 0 900 900 0 0
adjrange 1 0 0 900 900 0 0
adjrange 2 0 0 900 900 0 0
adjrange 3 0 0 900 900 0 0
adjrange 4 0 0 900 900 0 0
adjrange 5 0 0 900 900 0 0
adjrange 6 0 0 900 900 0 0
adjrange 7 0 0 900 900 0 0
adjrange 8 0 0 900 900 0 0
adjrange 9 0 0 900 900 0 0
adjrange 10 0 0 900 900 0 0
adjrange 11 0 0 900 900 0 0

# rxrange
rxrange 0 1000 2000
rxrange 1 1000 2000
rxrange 2 1000 2000
rxrange 3 1000 2000

# master
set looptime = 2000
set gyro_sync = OFF
set gyro_sync_denom = 2
set align_gyro = DEFAULT
set gyro_hardware_lpf = 42HZ
set gyro_lpf_hz = 60
set moron_threshold = 32
set gyro_notch1_hz = 0
set gyro_notch1_cutoff = 1
set gyro_notch2_hz = 0
set gyro_notch2_cutoff = 1
set align_acc = DEFAULT
set acc_hardware = MPU6050
set acc_lpf_hz = 15
set acczero_x = 228
set acczero_y = -54
set acczero_z = -377
set accgain_x = 4085
set accgain_y = 4072
set accgain_z = 3979
set rangefinder_hardware = NONE
set align_mag = CW180
set mag_hardware = NONE
set mag_declination = 0
set magzero_x = 8
set magzero_y = -102
set magzero_z = -91
set mag_calibration_time = 30
set baro_hardware = MS5611
set baro_use_median_filter = ON
set mid_rc = 1500
set min_check = 1100
set max_check = 1900
set rssi_channel = 0
set rssi_scale = 30
set rssi_invert = OFF
set rc_smoothing = ON
set serialrx_provider = SBUS
set sbus_inversion = ON
set spektrum_sat_bind = 0
set rx_min_usec = 885
set rx_max_usec = 2115
set blackbox_rate_num = 1
set blackbox_rate_denom = 1
set blackbox_device = SPIFLASH
set min_throttle = 1085
set max_throttle = 1850
set min_command = 1000
set motor_pwm_rate = 1000
set motor_pwm_protocol = ONESHOT125
set failsafe_delay = 5
set failsafe_recovery_delay = 5
set failsafe_off_delay = 200
set failsafe_throttle = 1000
set failsafe_throttle_low_delay = 100
set failsafe_procedure = SET-THR
set failsafe_stick_threshold = 50
set failsafe_fw_roll_angle = -200
set failsafe_fw_pitch_angle = 100
set failsafe_fw_yaw_rate = -45
set align_board_roll = -24
set align_board_pitch = -15
set align_board_yaw = 900
set gimbal_mode = NORMAL
set battery_capacity = 0
set vbat_scale = 110
set vbat_max_cell_voltage = 43
set vbat_min_cell_voltage = 33
set vbat_warning_cell_voltage = 35
set current_meter_scale = 400
set current_meter_offset = 0
set multiwii_current_meter_output = OFF
set current_meter_type = ADC
set yaw_motor_direction = 1
set yaw_jump_prevention_limit = 200
set 3d_deadband_low = 1406
set 3d_deadband_high = 1514
set 3d_neutral = 1460
set servo_center_pulse = 1500
set servo_pwm_rate = 50
set servo_lpf_hz = 20
set flaperon_throw_offset = 200
set tri_unarmed_servo = ON
set reboot_character = 82
set imu_dcm_kp = 2500
set imu_dcm_ki = 50
set imu_dcm_kp_mag = 10000
set imu_dcm_ki_mag = 0
set small_angle = 25
set fixed_wing_auto_arm = OFF
set disarm_kill_switch = ON
set auto_disarm_delay = 5
set gps_provider = UBLOX
set gps_sbas_mode = AUTO
set gps_dyn_model = AIR_1G
set gps_auto_config = ON
set gps_auto_baud = ON
set gps_min_sats = 6
set deadband = 5
set yaw_deadband = 5
set pos_hold_deadband = 20
set alt_hold_deadband = 50
set 3d_deadband_throttle = 50
set fw_autotune_overshoot_time = 100
set fw_autotune_undershoot_time = 200
set fw_autotune_threshold = 50
set fw_autotune_ff_to_p_gain = 10
set fw_autotune_ff_to_i_tc = 600
set inav_auto_mag_decl = ON
set inav_gravity_cal_tolerance = 5
set inav_use_gps_velned = OFF
set inav_gps_delay = 200
set inav_reset_altitude = FIRST_ARM
set inav_max_sonar_altitude = 200
set inav_w_z_sonar_p =  3.500
set inav_w_z_sonar_v =  6.100
set inav_w_z_baro_p =  0.350
set inav_w_z_gps_p =  0.200
set inav_w_z_gps_v =  0.500
set inav_w_xy_gps_p =  1.000
set inav_w_xy_gps_v =  2.000
set inav_w_z_res_v =  0.500
set inav_w_xy_res_v =  0.500
set inav_w_acc_bias =  0.010
set inav_max_eph_epv =  1000.000
set inav_baro_epv =  100.000
set nav_disarm_on_landing = OFF
set nav_use_midthr_for_althold = OFF
set nav_extra_arming_safety = OFF
set nav_user_control_mode = ATTI
set nav_position_timeout = 5
set nav_wp_radius = 100
set nav_wp_safe_distance = 10000
set nav_auto_speed = 300
set nav_auto_climb_rate = 500
set nav_manual_speed = 500
set nav_manual_climb_rate = 200
set nav_landing_speed = 200
set nav_land_slowdown_minalt = 500
set nav_land_slowdown_maxalt = 2000
set nav_emerg_landing_speed = 500
set nav_min_rth_distance = 500
set nav_rth_climb_first = ON
set nav_rth_climb_ignore_emerg = OFF
set nav_rth_tail_first = OFF
set nav_rth_allow_landing = ON
set nav_rth_alt_mode = AT_LEAST
set nav_rth_abort_threshold = 50000
set nav_rth_altitude = 1000
set nav_mc_bank_angle = 30
set nav_mc_hover_thr = 1500
set nav_mc_auto_disarm_delay = 2000
set nav_fw_cruise_thr = 1400
set nav_fw_min_thr = 1200
set nav_fw_max_thr = 1700
set nav_fw_bank_angle = 20
set nav_fw_climb_angle = 20
set nav_fw_dive_angle = 15
set nav_fw_pitch2thr = 10
set nav_fw_loiter_radius = 5000
set nav_fw_launch_velocity = 300
set nav_fw_launch_accel = 1863
set nav_fw_launch_detect_time = 40
set nav_fw_launch_thr = 1700
set nav_fw_launch_idle_thr = 1000
set nav_fw_launch_motor_delay = 500
set nav_fw_launch_spinup_time = 100
set nav_fw_launch_timeout = 5000
set nav_fw_launch_climb_angle = 18
set telemetry_switch = OFF
set telemetry_inversion = ON
set frsky_default_latitude =  0.000
set frsky_default_longitude =  0.000
set frsky_coordinates_format = 0
set frsky_unit = IMPERIAL
set frsky_vfas_precision = 0
set frsky_vfas_cell_voltage = OFF
set hott_alarm_sound_interval = 5
set smartport_uart_unidir = OFF
set ibus_telemetry_type = 0
set ltm_update_rate = NORMAL
set ledstrip_visual_beeper = OFF
set i2c_overclock = OFF
set debug_mode = NONE
set acc_task_frequency = 500
set attitude_task_frequency = 250
set async_mode = NONE
set throttle_tilt_comp_str = 0
set input_filtering_mode = OFF
set mode_range_logic_operator = OR

# profile
profile 1

set mc_p_pitch = 45
set mc_i_pitch = 30
set mc_d_pitch = 23
set mc_p_roll = 45
set mc_i_roll = 30
set mc_d_roll = 23
set mc_p_yaw = 85
set mc_i_yaw = 45
set mc_d_yaw = 0
set mc_p_level = 20
set mc_i_level = 15
set mc_d_level = 75
set fw_p_pitch = 5
set fw_i_pitch = 7
set fw_ff_pitch = 50
set fw_p_roll = 5
set fw_i_roll = 7
set fw_ff_roll = 50
set fw_p_yaw = 6
set fw_i_yaw = 10
set fw_ff_yaw = 60
set fw_p_level = 20
set fw_i_level = 5
set fw_d_level = 75
set max_angle_inclination_rll = 300
set max_angle_inclination_pit = 300
set dterm_lpf_hz = 40
set yaw_lpf_hz = 30
set dterm_setpoint_weight =  0.000
set fw_iterm_throw_limit = 165
set fw_reference_airspeed = 0
set dterm_notch_hz = 0
set dterm_notch_cutoff = 1
set pidsum_limit = 500
set yaw_p_limit = 300
set iterm_ignore_threshold = 200
set yaw_iterm_ignore_threshold = 50
set rate_accel_limit_roll_pitch = 0
set rate_accel_limit_yaw = 10000
set heading_hold_rate_limit = 90
set nav_mc_pos_z_p = 50
set nav_mc_pos_z_i = 0
set nav_mc_pos_z_d = 0
set nav_mc_vel_z_p = 100
set nav_mc_vel_z_i = 50
set nav_mc_vel_z_d = 10
set nav_mc_pos_xy_p = 65
set nav_mc_pos_xy_i = 120
set nav_mc_pos_xy_d = 10
set nav_mc_vel_xy_p = 180
set nav_mc_vel_xy_i = 15
set nav_mc_vel_xy_d = 100
set nav_fw_pos_z_p = 50
set nav_fw_pos_z_i = 0
set nav_fw_pos_z_d = 0
set nav_fw_pos_xy_p = 75
set nav_fw_pos_xy_i = 5
set nav_fw_pos_xy_d = 8
set rc_expo = 70
set rc_yaw_expo = 20
set thr_mid = 50
set thr_expo = 0
set roll_rate = 50
set pitch_rate = 60
set yaw_rate = 20
set tpa_rate = 0
set tpa_breakpoint = 1500

**Comment by shellixyz** [comment]
In your summary your were saying you are using 1.8.1 but it doesn't exist, only 1.8(.0). Then at the end of your first comment you are saying it is happening in iNav 1.9. But in your second comment it is written in your config dump that you are using version 1.7.0. I'm confused.

In your config dump you have no serial port configured with the GPS function. Wrong dump ?

**Comment by BadgerG** [comment]
I install 1.7 to see if the problem happens in a lower version. I had the same issue with 1.8.

**Comment by BadgerG** [comment]
I use the same port for GPS, so I have to unplug the GPS and MAG in order to down load using the USB. Otherwise it takes hours to get the data downloaded.

**Comment by BadgerG** [comment]
I explained all the in my first post.

**Comment by BadgerG** [comment]
Here is a video of what is happening. It is a Youtube video.

https://youtu.be/n0ytpiiqDig

**Comment by shellixyz** [comment]
Please post a config dump where you have the GPS configured. It is not a good idea to have the GPS on UART1. What are you using the other UARTs for ? My guess is that it is what is causing your issue.

**Comment by BadgerG** [comment]
I used the 3DR radio to get this dump so it tends to loose characters, so that is way I use the UBS to down load.

# version
# INAV/SPRACINGF3 1.7.0 May  3 2017 / 11:12:57 (104b66c)

# resources

# mixer
mixer QUADX

mmix reset


# servo
servo 0 1000 2000 1500 100 -1 
servo 1 1000 2000 1500 100 -1 
servo 2 1000 2000 1500 100 -1 
servo 3 1000 2000 1500 100 -1 
servo 4 1000 2000 1500 100 -1 
servo 5 1000 2000 1500 100 -1 
servo 6 1000 2000 1500 100 -1 
servo 7 1000 2000 1500 100 -1 

# servo mix
smix reset


# feature
feature -RX_PPM
feature -VBAT
feature -RX_SERIAL
feature -MOTOR_STOP
feature -SERVO_TILT
feature -SOFTSERIAL
feature -GPS
feature -TELEMETRY
feature -CURRENT_METER
feature -3D
feature -RX_PARALLEL_PWM
feature -RX_MSP
feature -RSSI_ADC
feature -LED_STRIP
feature -DASHBOARD
feature -BLACKBOX
feature -CHANNEL_FORWARDING
feature -TRANSPONDER
feature -AIRMODE
feature -SUPEREXPO
feature -VTX
feature -RX_SPI
feature -SOFTSPI
feature -PWM_SERVO_DRIVER
feature -PWM_OUTPUT_ENABLE
feature -OSD
feature VBAT
feature RX_SERIAL
feature GPS
feature TELEMETRY
feature BLACKBOX
feature PWM_OUTPUT_ENABLE

# beeper
beeper GYRO_CALIBRATED
beeper -HW_FAILURE
beeper RX_LOST
beeper RX_LOST_LANDING
beeper DISARMING
beeper ARMING
beeper ARMING_GPS_FIX
beeper BAT_CRIT_LOW
beeper BAT_LOW
beeper GPS_STATUS
beeper RX_SET
beeper ACTION_SUCCESS
beeper ACTION_FAIL
beeper READY_BEEP
beeper MULTI_BEEPS
beeper DISARM_REPEAT
beeper ARMED
beeper SYSTEM_INIT
beeper ON_USB
beeper LAUNCH_MODE

# map
map AETR1234

# serial
serial 0 2 115200 115200 0 115200
serial 1 1 57600 38400 0 115200
serial 2 64 115200 38400 0 115200

# led
led 0 0,0::C:0
led 1 0,0::C:0
led 2 0,0::C:0
led 3 0,0::C:0
led 4 0,0::C:0
led 5 0,0::C:0
led 6 0,0::C:0
led 7 0,0::C:0
led 8 0,0::C:0
led 9 0,0::C:0
led 10 0,0::C:0
led 11 0,0::C:0
led 12 0,0::C:0
led 13 0,0::C:0
led 14 0,0::C:0
led 15 0,0::C:0
led 16 0,0::C:0
led 17 0,0::C:0
led 18 0,0::C:0
led 19 0,0::C:0
led 20 0,0::C:0
led 21 0,0::C:0
led 22 0,0::C:0
led 23 0,0::C:0
led 24 0,0::C:0
led 25 0,0::C:0
led 26 0,0::C:0
led 27 0,0::C:0
led 28 0,0::C:0
led 29 0,0::C:0
led 30 0,0::C:0
led 31 0,0::C:0

# color
color 0 0,0,0
color 1 0,255,255
color 2 0,0,255
color 3 30,0,255
color 4 60,0,255
color 5 90,0,255
color 6 120,0,255
color 7 150,0,255
color 8 180,0,255
color 9 210,0,255
color 10 240,0,255
color 11 270,0,255
color 12 300,0,255
color 13 330,0,255
color 14 0,0,0
color 15 0,0,0

# mode_color
mode_color 0 0 1
mode_color 0 1 11
mode_color 0 2 2
mode_color 0 3 13
mode_color 0 4 10
mode_color 0 5 3
mode_color 1 0 5
mode_color 1 1 11
mode_color 1 2 3
mode_color 1 3 13
mode_color 1 4 10
mode_color 1 5 3
mode_color 2 0 10
mode_color 2 1 11
mode_color 2 2 4
mode_color 2 3 13
mode_color 2 4 10
mode_color 2 5 3
mode_color 3 0 8
mode_color 3 1 11
mode_color 3 2 4
mode_color 3 3 13
mode_color 3 4 10
mode_color 3 5 3
mode_color 4 0 7
mode_color 4 1 11
mode_color 4 2 3
mode_color 4 3 13
mode_color 4 4 10
mode_color 4 5 3
mode_color 5 0 9
mode_color 5 1 11
mode_color 5 2 2
mode_color 5 3 13
mode_color 5 4 10
mode_color 5 5 3
mode_color 6 0 6
mode_color 6 1 10
mode_color 6 2 1
mode_color 6 3 0
mode_color 6 4 0
mode_color 6 5 2
mode_color 6 6 3
mode_color 6 7 6
mode_color 6 8 0
mode_color 6 9 0
mode_color 6 10 0

# aux
aux 0 0 1 1300 2100
aux 1 1 0 1325 1600
aux 2 2 0 1850 2100
aux 3 20 1 1800 2100
aux 4 5 3 1300 1700
aux 5 0 0 900 900
aux 6 0 0 900 900
aux 7 0 0 900 900
aux 8 0 0 900 900
aux 9 0 0 900 900
aux 10 0 0 900 900
aux 11 0 0 900 900
aux 12 0 0 900 900
aux 13 0 0 900 900
aux 14 0 0 900 900
aux 15 0 0 900 900
aux 16 0 0 900 900
aux 17 0 0 900 900
aux 18 0 0 900 900
aux 19 0 0 900 900

# adjrange
adjrange 0 0 0 900 900 0 0
adjrange 1 0 0 900 900 0 900 900 0 0
adjrange 5 0 0 900 900 0 0
adjrange 6 0 0 900 900 0 0
adjrange 7 0 0 9e
rxrange 0 1000 2000
rxrange 1 1000 2000
rxrange 2 1000 2000
rxrange 3 1000 2000

# mastn_gyro = DEFAULT
set gyro_hardware_lpf = 42HZ
set gyro_lpf_hz = 60
set moron_threshold = 32
set gyro_notch1_hz = 0lign_acc = DEFAULT
set acc_hardware = MPU6050
set acc_lpf_hz = 15
set acczero_x = 228n_z = 3979
set rangefinder_hardware = NONE
set align_mag = CW180
set mag_hardware = HMC5883
set mag_declinatio_time = 30
set baro_hardware = MS5611
set baro_use_median_filter = ON
set mid_rc = 1500
set min_check = 1100
set  rc_smoothing = ON
set serialrx_provider = SBUS
set sbus_inversion = ON
set spektrum_= 1085
set max_throttle = 1850
set min_command = 1000
set motor_pwm_rat5
set failsafe_recovery_delay = 5
set failsafe_off_delay = 200
set failsafe_throttle = 1000
set failsafe_throttle_et failsafe_fw_roll_angle = -200
set failsafe_fw_pitch_angle = 100
set failsafe_fw_yaw_rgimbal_mode = NORMAL
set battery_capacity = 0
set vbat_scale = 110
set vbat_max_cell_voltage = 43
set vbat_minset current_meter_offset = 0
set multiwii_current_meter_output = OFF
set current_meter_type = ADC
set yaw_motor_dirband_high = 1514
set 3d_neutral = 1460
set servo_center_pulse = 1500
set servo_pwm_ra2500
set imu_dcm_ki = 50
set imu_dcm_kp_mag = 10000
set imu_dcm_ki_mag = 0
set small_angle rm_delay = 5
set gps_provider = UBLOX
set gps_sbas_mode = AUTO
set gps_dyn_model = AIR_1G
set gps_auto_config = ON
set pos_hold_deadband = 20
set alt_hold_deadband = 50
set 3d_deadband_throttle = 50
sold = 50
set fw_autotune_ff_to_p_gain = 10
set fw_autotune_ff_to_i_tc = 600
set inav_auto_mag_decl = ON
set inet inav_reset_altitude = FIRST_ARM
set inav_max_sonar_altitude = 200
set inav_w_z_sonar_p =  3.500
set inav_w_z_songps_v =  0.500
set inav_w_xy_gps_p =  1.000
set inav_w_xy_gps_v =  2.000
set inav_w_z = OFF
set nav_use_midthr_for_althold = OFF
set nav_extra_arming_safety 5
set nav_wp_radius = 100
set nav_wp_safe_distance = 10000
set nav_auto_speed = 300
set nav_auto_climb_rate = 5000
set nav_land_slowdown_minalt = 500
set nav_land_slowdown_maxalt = 2000
set nav_emerg_b_ignore_emerg = OFF
set nav_rth_tail_first = OFF
set nav_rth_allow_landing = ON
set nav_rth_alt_mode = AT_LEASgle = 30
set nav_mc_hover_thr = 1500
set nav_mc_auto_disarm_delay = 2000
set nav_fw_cruise_thr = 1400
set nav_fw_mangle = 20
set nav_fw_dive_angle = 15
set nav_fw_pitch2thr = 10
set nav_fw_loiter_rad 1700
set nav_fw_launch_idle_thr = 1000
set nav_fw_launch_motor_delay = 500
set nav_fw_launc_angle = 18
set telemetry_switch = OFF
set telemetry_inversion = ON
set frsky_default_latitude =  0.000
set frsky_
set frsky_vfas_precision = 0
set frsky_vfas_cell_voltage = OFF
set hott_alarm_sound_inRMAL
set ledstrip_visual_beeper = OFF
set i2c_overclock = OFF
set debug_mode = NONE
set acc_task_frequency = 5r = 0
set input_filtering_mode = OFF
set mode_range_logic_operator = OR

# profile
profile 1

set mc_p_pitch = t mc_d_roll = 23
set mc_p_yaw = 85
set mc_i_yaw = 45
set mc_d_yaw = 0
set mc_p_level50
set fw_p_yaw = 6
set fw_i_yaw = 10
set fw_ff_yaw = 60
set fw_p_levenation_rll = 300
set max_angle_inclination_pit = 300
set dterm_lpf_hz = 40
set yaw_lpf_hz = 30
set dterm_setpoint_m_notch_hz = 0
set dterm_notch_cutoff = 1
set pidsum_limit = 500
set yaw_p_limit = 300itch = 0
set rate_accel_limit_yaw = 10000
set heading_hold_rate_limit = 90
set nav_mc_pos_z_p = 50
set nav_mc_
set nav_mc_vel_z_d = 10
set nav_mc_pos_xy_p = 65
set nav_mc_pos_xy_i = 120
set nav_mc_pos_xy_d = 10
set nav_mc_ve50
set nav_fw_pos_z_i = 0
set nav_fw_pos_z_d = 0
set nav_fw_pos_xy_p = 75
set nav_fwset pitch_rate = 60
set yaw_rate = 20
set tpa_rate = 0
set tpa_breakpoint = 1500

**Comment by BadgerG** [comment]
I changed the port setup so that the GPS is using UART2 and the problem is still there. 3D lock with 18 sats and it still keeps loosing lock every 3-4 seconds, then it comes back almost immediately.

**Comment by shellixyz** [comment]
Maybe its caused by interference from your 3DR modem. Did you try without it ?

**Comment by BadgerG** [comment]
Using UART2 does work better, it doesn't totally lose GPS, but the error count on the GPS page is high. 1744 error out of 25000 messages.

**Comment by BadgerG** [comment]
Then there is no way to see if the GPS has a good lock without the 3DR radio, so no there is no way to do that. Can the USB be used while a battery is connected?

**Comment by shellixyz** [comment]
Yes you can use the USB port with a battery connected but to use the USB port you will need the GPS connected on UART2.

**Comment by BadgerG** [comment]
Okay I connected without the 3DR radio and the results are the same.

**Comment by shellixyz** [comment]
How is your GPS powered ? Are you sure its supply voltage is the right voltage and clean ?

**Comment by BadgerG** [comment]
The power is coming from the FC. I don't think the power is a problem. The UBEC supplies 1.5 amps to the FC, reciever, and one 4 led light.

**Comment by shellixyz** [comment]
Well I don't know what the issue could be. I never had any issue making M8N GPS modules communicate with iNav.

**Comment by digitalentity** [comment]
@BadgerG since the issue is opened for version 1.8.1 please provide a relevant dump.
Try also connecting your GPS unit to UART2 and configure accordingly.

**Comment by BadgerG** [comment]
The FW version is actual 1.7.  The GPS is already connected to UART2 and I have already posted a dump. The problem is not as bad using port 2. Lock is achieved more quickly. However it still looses lock every few second for about 1/2 second repeatedly.

**Comment by shellixyz** [comment]
Looks like it is linked to the HolyBro M8N GPS module. Another user has the same issue see #2840.

**Comment by Thieu-s** [comment]
I have exactly the same problem as in the video above. On 2 different crafts, both with an Omnibus F4 Pro (one original and one clone) and 2 different M8N GPS modules from HobbyKing. Tested on different firmware 1.8 and 1.9. I've tested litterly all posible settings for GPS... All the same...
Even with nothing else connected, just the fc with the gps, powered by usb or lipo, same result... 

In the upper left corner the osd shows the blinking sats...

https://www.youtube.com/watch?v=pl3PWt3aqgE
